pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -60,9 +61,10 @@ def on_trainer_init(
         self.trainer._is_data_prepared = False
 
     def get_profiled_train_dataloader(self, train_dataloader):
-        profiled_dl = self.trainer.profiler.profile_iterable(
-            enumerate(prefetch_iterator(train_dataloader)), "get_train_batch"
-        )
+        self.prefetcher = DataFetcher()
```

Comment:
why isn't this done inside the loop (similar to evaluation loop)?

Reply:
The next PR will add profiling, and I will move the logic from the evaluation loop to the data_connector too.